### ISSUE TITLE
Precompiled script plugins - `GradleProperties has not been loaded yet.`

### ISSUE THREAD

**Issue by gavra0** [gradle/gradle#14005]
Original issue: https://issuetracker.google.com/162515856
Gradle version: 6.6.0-rc-4

Using the [test.zip](https://github.com/gradle/gradle/files/5021871/test.zip) project, `:buildSrc:generatePrecompiledScriptPluginAccessors` task fails with:

```
Caused by: java.lang.IllegalStateException: GradleProperties has not been loaded yet.
        at org.gradle.initialization.DefaultGradlePropertiesController$NotLoaded.gradleProperties(DefaultGradlePropertiesController.java:74)
        at org.gradle.initialization.DefaultGradlePropertiesController$SharedGradleProperties.gradleProperties(DefaultGradlePropertiesController.java:59)
        at org.gradle.initialization.DefaultGradlePropertiesController$SharedGradleProperties.find(DefaultGradlePropertiesController.java:50)
        at org.gradle.api.internal.provider.sources.GradlePropertyValueSource.obtain(GradlePropertyValueSource.java:36)
        at org.gradle.api.internal.provider.sources.GradlePropertyValueSource.obtain(GradlePropertyValueSource.java:24)
        at org.gradle.api.internal.provider.DefaultValueSourceProviderFactory$LazilyObtainedValue.lambda$obtainValueForThe1stTime$0(DefaultValueSourceProviderFactory.java:344)
        at org.gradle.internal.Try.ofFailable(Try.java:39)
        at org.gradle.api.internal.provider.DefaultValueSourceProviderFactory$LazilyObtainedValue.obtainValueForThe1stTime(DefaultValueSourceProviderFactory.java:344)
        at org.gradle.api.internal.provider.DefaultValueSourceProviderFactory$LazilyObtainedValue.obtain(DefaultValueSourceProviderFactory.java:333)
        at org.gradle.api.internal.provider.DefaultValueSourceProviderFactory$ValueSourceProvider.calculateOwnValue(DefaultValueSourceProviderFactory.java:299)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.getOrNull(AbstractMinimalProvider.java:93)
        at com.android.build.gradle.options.ProjectOptions.readTestRunnerArgs(ProjectOptions.java:153)
        at com.android.build.gradle.options.ProjectOptions.<init>(ProjectOptions.java:62)
        at com.android.build.gradle.options.ProjectOptions.<init>(ProjectOptions.java:76)
        at com.android.build.gradle.internal.plugins.BasePlugin.createProjectServices(BasePlugin.java:846)
        at com.android.build.gradle.internal.plugins.BasePlugin.basePluginApply(BasePlugin.java:255)
        at com.android.build.gradle.internal.plugins.BasePlugin.lambda$apply$0(BasePlugin.java:245)
        at com.android.build.gradle.internal.crash.CrashReporting.runAction(crash_reporting.kt:27)
        at com.android.build.gradle.internal.plugins.BasePlugin.apply(BasePlugin.java:243)
        at com.android.build.gradle.internal.plugins.BasePlugin.apply(BasePlugin.java:138)
        at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
        at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:177)
        at org.gradle.api.internal.plugins.DefaultPluginManager.access$100(DefaultPluginManager.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:272)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.plugins.DefaultPluginManager.lambda$doApply$0(DefaultPluginManager.java:157)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:43)
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:156)
```

Is `GradlePropertiesController` supposed to load properties earlier?

**Comment by bamboo** [comment]
Thanks for the report, @gavra0.

This is likely happening because the plugins block of a precompiled script plugins is evaluated against a synthetic Gradle project with no access to Gradle properties.

**Comment by hameno** [comment]
It still fails for me with 6.6.1 and AGP 4.2.0-alpha08

**Comment by bamboo** [comment]
The fix will be in the upcoming 6.7 RC1.
